### PR TITLE
06 email

### PR DESCRIPTION
**Pull request by odykyi**

### CODE REVIEW

File: api/auth/email.client.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+// using Twilio SendGrid's v3 Node.js Library
+// https://github.com/sendgrid/sendgrid-nodejs
+const sgMail = require("@sendgrid/mail");
+require("dotenv").config();
+
+//Написано не очень но работает )
+function sendMail(email, vertificationToken) {
```
Comment (odykyi) [inline api/auth/email.client.js]: async function sendMail(email, vertificationToken) {

Comment (odykyi) [reply api/auth/email.client.js]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function

**Review by odykyi — COMMENTED** [review]
comments

File: api/auth/email.client.js (hunk 1 above)
Comment (IP777) [reply api/auth/email.client.js]: Исправил

File: api/auth/auth.controller.js [hunk 2]
```diff
@@ -24,12 +24,16 @@ async function registerUser(req, res, next) {
 
 	try {
 		const hashPass = await bcrypt.hash(password, saltRounds);
+		const verificationToken = uuid.v4();
+
+		sendMail(email, verificationToken);
```
Comment (odykyi) [inline api/auth/auth.controller.js]: await sendMail(email, verificationToken);

File: api/auth/auth.controller.js [hunk 3]
```diff
@@ -101,8 +105,31 @@ async function logoutUser(req, res, next) {
...
+	} catch (err) {
+		next(err);
+	}
+}
+
 module.exports = {
 	registerUser,
 	loginUser,
 	logoutUser,
+	vertificationMail,
```
Comment (odykyi) [inline api/auth/auth.controller.js]: vertification ???

File: api/auth/auth.router.js [hunk 4]
```diff
@@ -14,5 +14,6 @@ authRouter.post(
 );
 authRouter.post("/login", validator.loginUser, controller.loginUser);
 authRouter.post("/logout", token.authorize, controller.logoutUser);
+authRouter.get("/verify/:verificationToken", controller.vertificationMail);
```
Comment (odykyi) [inline api/auth/auth.router.js]: vertification ??

File: api/auth/email.client.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+// using Twilio SendGrid's v3 Node.js Library
+// https://github.com/sendgrid/sendgrid-nodejs
+const sgMail = require("@sendgrid/mail");
+require("dotenv").config();
+
+//Написано не очень но работает )
+async function sendMail(email, vertificationToken) {
```
Comment (odykyi) [inline api/auth/email.client.js:6]:
```javascript

async function sendMail(email, vertificationToken) {
	sgMail.setApiKey(process.env.SENDGRID_API_KEY);
	const msg = {
		to: email,
		from: process.env.MY_EMAIL,
		subject: "Sending with Twilio SendGrid is Fun",
		text: "and easy to do anywhere, even with Node.js",
		html: `<a href="http://localhost:3010/auth/verify/${vertificationToken}">Vertificate you token</a>`,
	};

	const [response] = await sgMail.send(msg);

        console.log(response);
        return response
	
}
```

**Review by odykyi — COMMENTED** [review]
comments

**Comment by odykyi** [comment]
@IP777 wait for fixing new comments

File: api/auth/auth.controller.js [hunk 6]
```diff
@@ -24,12 +24,16 @@ async function registerUser(req, res, next) {
 
 	try {
 		const hashPass = await bcrypt.hash(password, saltRounds);
+		const vertificationToken = uuid.v4();
+
+		sendMail(email, vertificationToken);
```
Comment (odykyi) [inline api/auth/auth.controller.js]: await sendMail(email, vertificationToken);

**Review by odykyi — COMMENTED** [review]
comments

**Comment by odykyi** [comment]
@IP777 wait for fix

File: api/auth/auth.controller.js (hunk 6 above)
Comment (IP777) [reply api/auth/auth.controller.js]: Исправил

**Pull request merged**